sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2761,7 +2759,7 @@ ur_result_t enqueueReadWriteHostPipe(const QueueImplPtr &Queue,
                                      const std::string &PipeName, bool blocking,
                                      void *ptr, size_t size,
                                      std::vector<ur_event_handle_t> &RawEvents,
-                                     const detail::EventImplPtr &OutEventImpl,
+                                     detail::event_impl *OutEventImpl,
```

Comment:
I can somewhat believe that in-parameters don't change ownership model, but modifying out-parameter and saying "it's unchanged, trust me" sounds very suspicious. Can you explain more?

Or maybe in other words, is that out-parameter even necessary?

Reply:
If `OutEventImpl` is not `nullptr` then this function calls `OutEventImpl->setHandle(UREvent);` on L2805.

The same is true for the `SetKernelParamsAndLaunch` function.

Reply:
Looks good. Thank you!
I just fixed one misprint: "**cand** the modified interfaces update" -> "**and** the modified interfaces updat"

Reply:
Yeah, sorry to be a pain, but mistakes can result in leaks/use-after-free and other flaky failures, so I want to be extra careful with this stuff. 

Once Sergey replies about `const` this is LGTM.